### PR TITLE
Increase path's max_length for ImportedFile model to 4096

### LINKED ISSUE #5061: Increase name and path's max_lenght for ImportedFile model
Some projects have some large filenames/nested directories.

https://github.com/rtfd/readthedocs.org/blob/ae991c4f131e2dd22fd4ce61deec381ae33f3aa4/readthedocs/projects/models.py#L895-L897

Any reason to not do that? Any suggested number? (In sentry the length of the failed file was 256).

Related sentry issue https://sentry.io/read-the-docs/readthedocs-org/issues/659636695/

### PR DESCRIPTION
**Pull request by dojutsu-user**
Closes #5061

### CODE REVIEW

File: readthedocs/projects/models.py [hunk 1]
```diff
@@ -1089,7 +1089,7 @@ class ImportedFile(models.Model):
     )
     name = models.CharField(_('Name'), max_length=255)
```
Comment (ericholscher) [inline readthedocs/projects/models.py]: Believe we want to update the `name` as well.

Comment (dojutsu-user) [reply readthedocs/projects/models.py]: What should be the optimal value for `max_length`. I didn't find that in the discussion of #5061

Comment (stsewd) [reply readthedocs/projects/models.py]: Name only saves the filename not the full path with directories, so the limitation for filenames is only `255`, so we are safe.

File: readthedocs/projects/models.py [hunk 2]
```diff
@@ -1089,7 +1089,7 @@ class ImportedFile(models.Model):
...
     slug = models.SlugField(_('Slug'))
-    path = models.CharField(_('Path'), max_length=255)
+    path = models.CharField(_('Path'), max_length=4096)
```
Comment (stsewd) [inline readthedocs/projects/models.py]: We should add a comment why 4096 is our magic number here

Comment (dojutsu-user) [reply readthedocs/projects/models.py]: Thank you.
I have added the comment and pushed the changes.

**Pull request merged**